packages/x-date-pickers/src/internals/utils/themeVar.ts

```diff
@@ -0,0 +1,11 @@
+import { Theme } from '@mui/material/styles';
+import type { CssVarsTheme, ThemeVars } from '@mui/material/styles/experimental_extendTheme';
+
+export type ExtendedTheme = Theme | CssVarsTheme;
+
+export const getThemePalette = (theme: ExtendedTheme): Theme['palette'] | ThemeVars['palette'] => {
```

Comment:
I do like this solution a lot

Reply:
@siriwatknp If you're good with this solution I will do the same in the data-grid

Reply:
<img width="540" alt="Screen Shot 2565-11-16 at 11 23 48" src="https://user-images.githubusercontent.com/18292247/202083158-f925d565-fa01-471e-97aa-dd635aaeb4b3.png">

Seems to work.

Reply:
Confirmed, you need to add it to `tsconfig.build.json` too, otherwise, `yarn build` fails.

Reply:
clearly less hacky, and I confirm `yarn release:build` succeed

Reply:
Seems like apps using `@mui/x-date-pickers` with this approach also works and don't have access to `theme.vars`